bindings/java/src/junit/com/apple/foundationdb/tuple/TupleSerializationTest.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * TupleSerializationTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb.tuple;
+
+import java.math.BigInteger;
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import com.apple.foundationdb.FDBLibraryRule;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests about serializing and deserializing Tuples.
```

Comment:
It's a little unclear to me what the difference between a "tuple encoding test" and a "tuple serialization test". Is there a way you could elaborate the difference in one of these two test files?

Reply:
it's not super clear to me either :) 

Actually, it's the names that really are ambiguous. The two classes were separated to avoid having a 1000+ line test file that's hard for humans to parse, but the distinction isn't really about serialization, and more about the type of tests they are. I am open to alternative names...

Reply:
Interesting. It might just be good to document that somehow. Like, "This test and `TupleEncodingTest` contain similar tests but are broken up to avoid either file getting too large." It would also be kind of nice if there were a rule of thumb to aid test authors when they are trying to figure out which one to use.

Reply:
I renamed one to `TuplePackingTest`, which is marginally better, and added documentation